packages/webhooks/lib/utils.ts

```diff
@@ -0,0 +1,136 @@
+import crypto from 'crypto';
+import { backOff } from 'exponential-backoff';
+import type { AxiosError } from 'axios';
+import { axiosInstance as axios } from '@nangohq/utils';
+import type { Environment } from '@nangohq/types';
+import type { LogContext } from '@nangohq/logs';
+
+export const RETRY_ATTEMPTS = 7;
+
+export const NON_FORWARDABLE_HEADERS = [
+    'host',
+    'authorization',
+    'connection',
+    'keep-alive',
+    'content-length',
+    'content-type', // we're sending json so don't want this overwritten
+    'content-encoding',
+    'cookie',
+    'set-cookie',
+    'referer',
+    'user-agent',
+    'sec-',
+    'proxy-',
+    'www-authenticate',
+    'server'
+];
+
+export const retry = async (
+    activityLogId: number | null,
+    logCtx?: LogContext | null | undefined,
+    error?: AxiosError,
+    attemptNumber?: number
+): Promise<boolean> => {
+    if (error?.response && (error?.response?.status < 200 || error?.response?.status >= 300)) {
+        const content = `Webhook response received an ${
+            error?.response?.status || error?.code
+        } error, retrying with exponential backoffs for ${attemptNumber} out of ${RETRY_ATTEMPTS} times`;
+
+        if (activityLogId) {
+            await logCtx?.error(content);
+        }
+
+        return true;
+    }
+
+    return false;
+};
+
+export const getSignatureHeader = (secret: string, payload: unknown): Record<string, string> => {
+    const combinedSignature = `${secret}${JSON.stringify(payload)}`;
+    const createdHash = crypto.createHash('sha256').update(combinedSignature).digest('hex');
+
+    return {
+        'X-Nango-Signature': createdHash
+    };
+};
+
+export const filterHeaders = (headers: Record<string, string>): Record<string, string> => {
+    const filteredHeaders: Record<string, string> = {};
+
+    for (const [key, value] of Object.entries(headers)) {
+        if (NON_FORWARDABLE_HEADERS.some((header) => key.toLowerCase().startsWith(header))) {
+            continue;
+        }
+
+        filteredHeaders[key] = value;
+    }
+
+    return filteredHeaders;
+};
+
+export const shouldSend = (environment: Environment, options?: { auth?: boolean; forward?: boolean }): boolean => {
+    const hasAnyWebhook = environment.webhook_url || environment.webhook_url_secondary;
+
+    if (options?.forward && hasAnyWebhook) {
+        return true;
+    }
+
+    const authNotSelected = options?.auth && !environment.send_auth_webhook;
+
+    if (!hasAnyWebhook || authNotSelected) {
+        return false;
+    }
+
+    return true;
```

Comment:
cannot we simplify it like this: 
```
  if (!hasAnyWebhook) {
    return false;
  }
  
  if (options?.forward) {
    return true;
  }
  
  if (options?.auth && !environment.send_auth_webhook) {
    return false;
  }
  
  return true;
```

Reply:
This whole condition is going to change with the additional notifications so will simplify this later.